strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/layout/npm/NpmArtifactController.java

```diff
@@ -451,6 +494,116 @@ public ResponseEntity addUser(Authentication authentication)
                        .body("{\"ok\":\"user '" + authentication.getName() + "' created\"}");
     }
 
+    @DeleteMapping(path = "{storageId}/{repositoryId}/{packageScope}/{packageName}/-rev/{rev}")
+    @PreAuthorize("hasAuthority('ARTIFACTS_DELETE')")
+    public ResponseEntity unpublishPackageWithScope(@RepositoryMapping Repository repository,
+                                                    @PathVariable(name = "packageScope") String packageScope,
+                                                    @PathVariable(name = "packageName") String packageName,
+                                                    @PathVariable(name = "rev") String rev)
+    {
+
+        logger.info("Npm unpublish a package request: storageId-[{}]; repositoryId-[{}]; packageName-[{}]; revision-[{}];",
+                repository.getStorage().getId(), repository.getId(), packageName, rev);
+
+        NpmUnpublishService.Result result = npmUnpublishService.unpublishPackage(repository, packageScope, packageName);
+
+        return processUnpublishResult(result);
+    }
+
+    /**
+     * Unpublish a single version of a specified package. This mapping works for npm versions > 6.5.0.
+     *
+     * @param repository   repository
+     * @param packageScope package scope
+     * @param packageName  package name
+     * @param tarball      tarball
+     * @param rev          revision value
+     * @return result via {@link ResponseEntity} with HTTP status
+     * @throws Exception
+     */
+    @DeleteMapping(path = "{storageId}/{repositoryId}/{r1}/{r2}/{r3}/{packageScope}/{packageName}/-/{tarball}/-rev/{rev}")
+    @PreAuthorize("hasAuthority('ARTIFACTS_DELETE')")
+    public ResponseEntity unpublishVersionWithScope(@RepositoryMapping Repository repository,
+                                                    @PathVariable(name = "packageScope") String packageScope,
+                                                    @PathVariable(name = "packageName") String packageName,
+                                                    @PathVariable(name = "tarball") String tarball,
+                                                    @PathVariable(name = "rev") String rev)
+    {
+
+        final String version = getPackageVersion(tarball, packageName)
+                                       .replace(".tgz", "");
+
+        logger.info("Npm unpublish a single version request: storageId-[{}]; repositoryId-[{}]; packageName-[{}]; tarball-[{}]; revision-[{}];",
+                repository.getStorage().getId(), repository.getId(), packageName, tarball, rev);
+
+        NpmUnpublishService.Result result = npmUnpublishService.unpublishSingleVersion(repository, packageScope,
+                                                                                       packageName, tarball,
+                                                                                       version);
```

Comment:
Cosmetic: Would you mind re-aligning this and having one parameter per line here? Thanks! :)